src/main/realms.c

```diff
@@ -2371,9 +2372,10 @@ int realms_init(CONF_SECTION *config)
 			}
 			if (*p != '\0') continue;
 		
-			if (home_server_afrom_file(dp->d_name) < 0) {
+			snprintf(conf_file, sizeof(conf_file), "%s/%s", rc->directory, dp->d_name);
+			if (home_server_afrom_file(conf_file) < 0) {
 				ERROR("Failed reading home_server from %s - %s",
-				      dp->d_name, fr_strerror());
+				      conf_file, fr_strerror());
```

Comment:
you can just skip the snprintf, and do"

```
ERROR("Failed reading home_server from %s /%s - %s",
   rc->directory, dp->d_name, fr_strerror());
```

less code is simpler.

Reply:
we need the snprintf() to have the fullpath needed by `home_server_afrom_file()`